### ISSUE TITLE
If a task refers to an input with a space in it, conductor cannot schedule it, throws exception

### ISSUE THREAD

**Issue by Jiehong** [Netflix/conductor#879]
Hello,

If you create the following workflow:

```
{
  "type": "workflow-definition",
  "label": "My Label Here",
  "name": "WKF_NAME",
  "description": "Example workflow",
  "inputParameters": {
  },
  "tasks": [{
    "taskType": "task_type_here",
    "taskReferenceName": "ref1",
    "label": "Record",
    "description": "This task does nothing",
    "inputMapping": {
      "blabla": "${workflow.input.Space Value}"
    },
    "startDelaySeconds": 0
  }]
}
```

If you start this workflow, the execution will fail right away, as conductor will try to schedule the only task, and tries to replace the data in the inputMapping of the task.

This generates the following exception:

```
 com.jayway.jsonpath.InvalidPathException: Use bracket notion ['my prop'] if your property contains blank characters. position: 17
at com.jayway.jsonpath.internal.path.PathCompiler.readPropertyOrFunctionToken(PathCompiler.java:184)
at com.jayway.jsonpath.internal.path.PathCompiler.readNextToken(PathCompiler.java:145)
at com.jayway.jsonpath.internal.path.PathCompiler.readDotToken(PathCompiler.java:165)
at com.jayway.jsonpath.internal.path.PathCompiler.readNextToken(PathCompiler.java:139)
at com.jayway.jsonpath.internal.path.PathCompiler.readPropertyOrFunctionToken(PathCompiler.java:231)
at com.jayway.jsonpath.internal.path.PathCompiler.readNextToken(PathCompiler.java:145)
at com.jayway.jsonpath.internal.path.PathCompiler.readDotToken(PathCompiler.java:165)
at com.jayway.jsonpath.internal.path.PathCompiler.readNextToken(PathCompiler.java:139)
at com.jayway.jsonpath.internal.path.PathCompiler.readPropertyOrFunctionToken(PathCompiler.java:231)
at com.jayway.jsonpath.internal.path.PathCompiler.readNextToken(PathCompiler.java:145)
at com.jayway.jsonpath.internal.path.PathCompiler.readDotToken(PathCompiler.java:165)
at com.jayway.jsonpath.internal.path.PathCompiler.readNextToken(PathCompiler.java:139)
at com.jayway.jsonpath.internal.path.PathCompiler.readContextToken(PathCompiler.java:118)
at com.jayway.jsonpath.internal.path.PathCompiler.compile(PathCompiler.java:54)
at com.jayway.jsonpath.internal.path.PathCompiler.compile(PathCompiler.java:69)
at com.jayway.jsonpath.JsonPath.<init>(JsonPath.java:101)
at com.jayway.jsonpath.JsonPath.compile(JsonPath.java:467)
at com.jayway.jsonpath.internal.JsonContext.read(JsonContext.java:149)
at com.netflix.conductor.core.execution.ParametersUtils.replaceVariables(ParametersUtils.java:217)
at com.netflix.conductor.core.execution.ParametersUtils.replace(ParametersUtils.java:167)
at com.netflix.conductor.core.execution.ParametersUtils.getTaskInputV2(ParametersUtils.java:128)
at com.netflix.conductor.core.execution.ParametersUtils.getTaskInput(ParametersUtils.java:63)
at com.netflix.conductor.core.execution.DeciderService.getTasksToBeScheduled(DeciderService.java:443)
at com.netflix.conductor.core.execution.DeciderService.getTasksToBeScheduled(DeciderService.java:437)
at com.netflix.conductor.core.execution.DeciderService.getNextTask(DeciderService.java:308)
at com.netflix.conductor.core.execution.DeciderService.decide(DeciderService.java:172)
at com.netflix.conductor.core.execution.DeciderService.decide(DeciderService.java:99)
at com.netflix.conductor.core.execution.WorkflowExecutor.decide(WorkflowExecutor.java:575)
at com.netflix.conductor.core.execution.WorkflowExecutor.updateTask(WorkflowExecutor.java:531)
at com.netflix.conductor.service.ExecutionService.updateTask(ExecutionService.java:162)
```

This happens anytime conductor tries to schedule a task with a reference to a variable having a space in it (whether this variable exists or not).

This has been tested with conductor 1.10.17.

Instead, it should rather do the following:
- check if a space is present (or catch the underlying exception);
- set status FAILED to the task referencing this bad variable;
- add a message as to why the task is failed (referencing variable with a space-name).

**Comment by jwulf** [comment]
Is there a validator? Seems like this would be a trivial PR.....

**Comment by jwulf** [comment]
Maybe this PR? https://github.com/Netflix/conductor/pull/862

**Comment by cyzhao** [comment]
@falu2010-netflix please check if this is handled by validation PR, if not let's improve as needed.

**Comment by apanicker-nflx** [comment]
This should be fixed as part of #862. Closing this issue, please reopen if you still see this.
